src/sunburst/index.js

```diff
@@ -0,0 +1,133 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import React, {PropTypes} from 'react';
+import {
+  hierarchy,
+  partition
+} from 'd3-hierarchy';
+
+import {
+  scaleLinear,
+  scaleSqrt
+} from 'd3-scale';
+
+import {AnimationPropType} from 'animation';
+import ArcSeries from 'plot/series/arc-series';
+import XYPlot from 'plot/xy-plot';
+
+function getRadialDomain(data) {
+  return data.reduce((res, row) => Math.max(row.radius, res), 0);
+}
+
+function cleanParent(node) {
+  if (!node.parent) {
+    return {...node};
+  }
+  return {
+    ...node,
+    parent: cleanParent(node.parent)
+  };
+}
+
+class Sunburst extends React.Component {
+
+  static get propTypes() {
+    return {
+      animation: AnimationPropType,
+      className: PropTypes.string,
+      data: PropTypes.object.isRequired,
+      height: PropTypes.number.isRequired,
+      hideRootNode: PropTypes.bool,
+      onValueClick: PropTypes.func,
+      onValueMouseOver: PropTypes.func,
+      onValueMouseOut: PropTypes.func,
+      width: PropTypes.number.isRequired
+    };
+  }
+
+  static get defaultProps() {
+    return {
+      className: ''
+    };
+  }
+
+  /**
+   * Create the list of nodes to render.
+   * @returns {Array} Array of nodes.
+   */
+  _getNodesToRender() {
+    const {data, height, hideRootNode, width} = this.props;
+    const partitionFunction = partition();
+    const structuredInput = hierarchy(data).sum(d => d.size);
+    const radius = (Math.min(width, height) / 2) - 10;
+    const x = scaleLinear().range([0, 2 * Math.PI]);
+    const y = scaleSqrt().range([0, radius]);
+
+    return partitionFunction(structuredInput).descendants()
+      .reduce((res, cell, index) => {
+        if (hideRootNode && index === 0) {
+          return res;
+        }
+
+        return res.concat([{
+          angle0: Math.max(0, Math.min(2 * Math.PI, x(cell.x0))),
+          angle: Math.max(0, Math.min(2 * Math.PI, x(cell.x1))),
+          radius0: Math.max(0, y(cell.y0)),
+          radius: Math.max(0, y(cell.y1)),
```

Comment:
that'd be following the d3 naming convention, not the react-vis one. There is not a notion of {attr}1, there is only {attr} and {attr}0.